test/cpp/api/serialize.cpp

```diff
@@ -60,6 +60,25 @@ TEST(SerializeTest, BasicToFile) {
   ASSERT_TRUE(x.allclose(y));
 }
 
+TEST(SerializeTest, BasicViaFunc) {
+  torch::manual_seed(0);
+
+  auto x = torch::randn({5, 5});
+
+  std::string serialized;
+  torch::save(x, [&](const void* buf, size_t n) {
+    serialized.append(reinterpret_cast<const char *>(buf), n);
+    return n;
+  });
+  std::istringstream readHelper(serialized);
+  torch::Tensor y;
+  torch::load(y, readHelper);
```

Comment:
For API orthogonality, we should probably make the same changes to `load` as well.

Reply:
Yes, totally agree.
My intention was partly to gauge whether there would be objections that I hadn't thought of to this type of approach before doing the parallel load path. But I'll go ahead and change load as well.